### PR TITLE
self-hosted functions

### PR DESCRIPTION
**Pull request by kiwicopple**

### CODE REVIEW

File: spec/functions_v0_config.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      This is will seed a singular user into the database, and will disable browser authentication. All browser usage will default to this user. Inviting team users and other team-related functionality is currently not supported for self-hosted. Logflare self-hosted is currently intended for single-user experience only.
+  - id: 'LOGFLARE_API_KEY' # {string} A unique identifier for this param.
+    title: 'LOGFLARE_API_KEY' # {string} Any name.
+    tags: ['general'] # {string[]} These tags are useful for grouping parameters
+    links: [] # {string[]} These tags are useful for grouping parameters
+    required: true
+    default: ''
+    type: 'string'
+    description: |
+      Allows you to pass in an API key that will used for authentication. This is intended for programmatic usage where an external program sets the API key. It is advised to use the UI to configure the access tokens instead. If this value is not provided, the default API key will be automatically generated.
```
Comment (gregnr) [inline spec/functions_v0_config.yaml:37]:
```suggestion
      Allows you to pass in an API key that will be used for authentication. This is intended for programmatic usage where an external program sets the API key. It is advised to use the UI to configure the access tokens instead. If this value is not provided, the default API key will be automatically generated.
```

Comment (kiwicopple) [reply spec/functions_v0_config.yaml:37]: ah - this is all from Analytics actually. I'm just stubbing out the navigation

Comment (gregnr) [reply spec/functions_v0_config.yaml:37]: Got it!

File: apps/docs/docs/ref/self-hosting-functions/introduction.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+```bash
+docker build -t edge-runtime .
+docker run -it --rm -p 9000:9000 -v /path/to/supabase/functions:/usr/services supabase/edge-runtime start --main-service /usr/services
+```
+
+## How to update to a newer Deno version
+
+- Select the Deno version to upgrade and visit its tag on GitHub (eg: https://github.com/denoland/deno/blob/v1.30.3/Cargo.toml)
+- Open the `Cargo.toml` at the root of of this repo and modify all `deno_*` modules to match to the selected tag of Deno.
```
Comment (gregnr) [inline apps/docs/docs/ref/self-hosting-functions/introduction.mdx:45]:
```suggestion
- Open the `Cargo.toml` at the root of this repo and modify all `deno_*` modules to match to the selected tag of Deno.
```

**Review by inian — APPROVED** [review]

**Review by MildTomato — APPROVED** [review]

**Pull request merged**